### PR TITLE
Feature: Implement RENAME group in meatball menu in mobile header

### PR DESCRIPTION
**Pull request by ShoeheyOt**
## Overview
Implement renaming group function in group single page

## Screen Captures

[rename-groupSingle.webm](https://github.com/nishiki-tech/nishiki-frontend/assets/142355969/9fc221f4-72c0-44fc-a25d-3677d1384619)

## Assignee Checklist:

- [X] The naming convention of the PR title is correct (See the comment at the top of this template)
- [X] The base branch is correct (See: [Types of Branches](https://github.com/nishiki-tech/nishiki-frontend/blob/develop/docs/CONTRIBUTING.md#types-of-branches))
- [X] The branch name follows the [Branch Naming Conventions](https://github.com/nishiki-tech/nishiki-frontend/blob/develop/docs/CONTRIBUTING.md#branch-naming-conventions)
- [X] The correct assignees and reviewers have been designated for this PR
- [X] The coding style follows the [Coding Style Guide](https://github.com/nishiki-tech/nishiki-frontend/blob/develop/docs/STYLEGUIDE.md)
- [X] All the related issues are associated with this PR
- [X] All criteria in the associated issues are met (please tick the checkboxes)
- [X] My changes do not generate new warnings or errors (especially in the console)

## Reviewer Checklist:

- [ ] Double-check the "Assignee Checklist"
- [ ] The code follows the generic best practices and our [Coding Style Guide](https://github.com/nishiki-tech/nishiki-frontend/blob/develop/docs/STYLEGUIDE.md)
- [ ] The code is well-commented and easy to understand
- [ ] The UI changes accurately reflect the provided design

### CODE REVIEW

File: src/components/pages/GroupSinglePage.tsx
Comment (ShoeheyOt) [inline src/components/pages/GroupSinglePage.tsx:1]: Please ignoe this file as the change here is only adding props to be passed

File: src/features/groups/components/GroupSingleHeaderDropdownMenuTriggerButton/GroupSingleHeaderDropdownMenuContent/index.tsx
Comment (ShoeheyOt) [inline src/features/groups/components/GroupSingleHeaderDropdownMenuTriggerButton/GroupSingleHeaderDropdownMenuContent/index.tsx:1]: Please ignore this file as the changes are only passing props

File: src/features/groups/components/GroupSingleHeaderDropdownMenuTriggerButton/index.tsx
Comment (ShoeheyOt) [inline src/features/groups/components/GroupSingleHeaderDropdownMenuTriggerButton/index.tsx:1]: Please ignore this file as well

**Review by ShoeheyOt — COMMENTED** [review]
@nick-y-ito @kanta1207 Hi guys. could you please review this PR when you have time??
Thanks :)

**Comment by nick-y-ito** [comment]
@ShoeheyOt 

I found a serious issue. When the group renaming drawer is open, the [ space ] and [ enter/return ] keys do not work. I guess it's being caused by the dropdown menu remaining in the background. Because the [ space ] and [ enter/return ] keys also have some functionality on the dropdown menu, the functionalities of the drawer and the dropdown menu are conflicting.

In the video, I pressed the [ space ] and [ return ] keys on both the input and buttons, but nothing happened. I also tried the same thing on my iPhone, but it caused the same issue.

I suppose the best solution is to avoid nesting the dropdown and the drawer.

https://github.com/nishiki-tech/nishiki-frontend/assets/99148565/45e6b047-ea72-4f8f-9494-83ee2fa1fa44

**Comment by nick-y-ito** [comment]
@kanta1207
Could you, too, take a look at my comment below, please?

**Comment by kanta1207** [comment]
> @kanta1207 I hope you will take a look at my comment above and share your opinion as well.

I tried in my environment and found the same issue. 

I tried to find some ways to solve this issue, but couldn't. 
Indeed, it looks like we need to restructure all of 
DropdownMenu + nested Drawer/Dialog to resolve this issue.

**Comment by nick-y-ito** [comment]
@kanta1207 @ShoeheyOt 
Thank you for your investigation. I suppose we should separate this issue https://github.com/nishiki-tech/nishiki-frontend/pull/343#issuecomment-1986922413.

**Review by nick-y-ito — CHANGES_REQUESTED** [review]
@ShoeheyOt 

Thank you for your implementation! Please kindly take a look at the comments.

In addition, there has to be a gap on the right of the dropdown menu.

image

**Review by nick-y-ito — APPROVED** [review]
@ShoeheyOt LGTM! Thank you for fixing them!

**Pull request merged**